### PR TITLE
[WIP] Error percentage monitor now checks error per field

### PR DESCRIPTION
**Pull request by leovictorsr**
This should implement what's asked on issue #151 .

Currently I am looking for a way to have multiple fields with errors being logged without having to refactor much of the current code.
Tried with subtests but I would have to change more of what's already changed, I'll implement it after discussing the best way with our maintainers.

### CODE REVIEW

**Comment by leovictorsr** [comment]
@rennerocha @ejulio please review.

File: spidermon/contrib/monitors/mixins/validation.py [hunk 1]
```diff
@@ -120,30 +122,34 @@ def _check_missing_required_count(self, missing_count, allowed_count):
...
+
+    def _check_missing_required_percent(self, missing_percent, field_name, allowed_percent=0):
+        try:
+            self.assertLessEqual(
+                missing_percent,
+                allowed_percent,
+                msg="{percent}% of required field {field_name} are missing!{threshold_info}".format(
+                    percent=missing_percent * 100,
+                    field_name=field_name,
+                    threshold_info=(" (maximum allowed %.0f%%)" % (allowed_percent * 100))
```
Comment (ejulio) [inline spidermon/contrib/monitors/mixins/validation.py]: Maybe using a variable is a better option here, too much code in one line

File: spidermon/contrib/monitors/mixins/validation.py [hunk 2]
```diff
@@ -120,30 +122,34 @@ def _check_missing_required_count(self, missing_count, allowed_count):
...
-            msg="{percent}% of required fields are missing!{threshold_info}".format(
-                percent=missing_percent * 100,
-                threshold_info=(" (maximum allowed %.0f%%)" % (allowed_percent * 100))
-                if allowed_percent > 0
-                else "",
-            ),
-        )
+        self._check_missing_required_percent(missing_percent, field_name, allowed_percent)
+
+    def _check_missing_required_percent(self, missing_percent, field_name, allowed_percent=0):
```
Comment (ejulio) [inline spidermon/contrib/monitors/mixins/validation.py]: Maybe `field_name` should be the first argument, keep the percentages together

File: spidermon/contrib/monitors/mixins/validation.py [hunk 3]
```diff
@@ -193,16 +204,20 @@ def check_field_errors_percent(self, field_name, errors=None, allowed_percent=0)
...
-            msg="{percent}% of fields have validation errors!{threshold_info}".format(
-                percent=errors_percent * 100,
-                threshold_info=(" (maximum allowed %.0f%%)" % (allowed_percent * 100))
-                if allowed_percent > 0
-                else "",
-            ),
-        )
+        self._check_field_errors_percent(errors_percent, field_name, allowed_percent)
+
+    def _check_field_errors_percent(self, errors_percent, field_name, allowed_percent):
```
Comment (ejulio) [inline spidermon/contrib/monitors/mixins/validation.py]: `_check_field_errors_percent` and `_check_missing_required_percent` are quite similar, maybe extract a base function for both

File: spidermon/contrib/monitors/mixins/validation.py [hunk 4]
```diff
@@ -193,16 +204,20 @@ def check_field_errors_percent(self, field_name, errors=None, allowed_percent=0)
...
+                msg="{percent}% of field {field_name} have validation errors!{threshold_info}".format(
+                    percent=errors_percent * 100,
+                    field_name=field_name,
+                    threshold_info=(" (maximum allowed %.0f%%)" % (allowed_percent * 100))
+                    if allowed_percent > 0
+                    else "",
+                ),
+            )
+        except AssertionError:
+            traceback.print_exc()
```
Comment (ejulio) [inline spidermon/contrib/monitors/mixins/validation.py]: Not sure why this line.
Do you intend to log the error? If true, then why not `logger.exception` or `logger.error`?

**Comment by rennerocha** [comment]
This issue was solved in https://github.com/scrapinghub/spidermon/pull/164.

**Pull request closed without merging**